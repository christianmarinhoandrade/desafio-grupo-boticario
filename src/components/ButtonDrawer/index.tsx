import { Button, ButtonProps } from 'antd'
import { MenuOutlined } from '@ant-design/icons'

export type ButtonDrawerProps = { onClick?: () => void } & ButtonProps
const ButtonDrawer = ({ onClick, ...props }: ButtonDrawerProps) => {
  return <Button onClick={onClick} icon={<MenuOutlined />} {...props} />
}

export default ButtonDrawer
