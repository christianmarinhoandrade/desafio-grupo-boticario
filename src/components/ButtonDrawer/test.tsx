import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import ButtonDrawer from '.'

describe('<ButtonDrawer />', () => {
  it('should render button drawer default', () => {
    const { container } = render(<ButtonDrawer />)

    expect(screen.getByRole('img')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render button drawer to call the method if clicked', async () => {
    const onClick = jest.fn()
    render(<ButtonDrawer onClick={onClick} />)

    const button = screen.getByRole('button')

    userEvent.click(button)
    await waitFor(() => {
      expect(onClick).toHaveBeenCalledTimes(1)
    })
  })
})
