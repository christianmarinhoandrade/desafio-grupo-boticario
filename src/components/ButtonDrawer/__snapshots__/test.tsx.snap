// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonDrawer /> should render button drawer default 1`] = `
<button
  class="ant-btn ant-btn-icon-only"
  type="button"
>
  <span
    aria-label="menu"
    class="anticon anticon-menu"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="menu"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
      />
    </svg>
  </span>
</button>
`
